plasmapy/utils/roman/roman.py

```diff
@@ -0,0 +1,213 @@
+    """, re.VERBOSE)
+
+
+def to_roman(n: typing.Union[int, np.integer]) -> str:
+    """
+    Convert an integer to a Roman numeral.
+
+    Parameters
+    ----------
+    n : int or `~numpy.integer`
+        The integer to be converted to a Roman numeral that must be
+        between 1 and 4999, inclusive.
+
+    Returns
+    -------
+    result : str
+        The number in Roman numeral notation.
+
+    Raises
+    ------
+    TypeError
+        If the input is not an integer.
+
+    ~plasmapy.utils.roman.OutOfRangeError
+        If the number is not between 1 and 4999, inclusive.
+
+    See Also
+    --------
+    from_roman
+
+    Examples
+    --------
+    >>> to_roman(5)
+    'V'
+    >>> to_roman(2525)
+    'MMDXXV'
+
+    """
+    if not isinstance(n, (int, np.integer)):
+        raise TypeError(f"{n} cannot be converted to a Roman numeral.")
+    if not (0 < n < 5000):
+        raise OutOfRangeError("Number is out of range (need 0 < n < 5000)")
+
+    result = ""
+    for numeral, integer in _romanNumeralMap:
+        while n >= integer:
+            result += numeral
+            n -= integer
+    return result
+
+
+def from_roman(s: str) -> int:
+    """
+    Convert a Roman numeral to an integer.
+
+    Parameters
+    ----------
+    s : str
+        A Roman numeral.
+
+    Returns
+    -------
+    result : int
+        The integer corresponding to the Roman numeral.
+
+    Raises
+    ------
+    TypeError
+        The argument is not a `str`.
+
+    ~plasmapy.utils.roman.InvalidRomanNumeralError
+        The argument is not a valid Roman numeral.
+
+    See Also
+    --------
+    to_roman
+
+    Examples
+    --------
+    >>> from_roman('V')
+    5
+    >>> from_roman('MMMMCCCLXVII')
+    4367
+
+    """
+    if not isinstance(s, str):
+        raise TypeError('The argument to from_roman must be a string.')
+    if not _romanNumeralPattern.search(s):
+        raise InvalidRomanNumeralError('Invalid Roman numeral: %s' % s)
+
+    result = 0
+    index = 0
+    for numeral, integer in _romanNumeralMap:
+        while s[index:index+len(numeral)] == numeral:
+            result += integer
+            index += len(numeral)
+    return result
+
+
+def is_roman_numeral(s: str) -> bool:
+    """
+    Check whether or not a string is a valid Roman numeral.
+
+    Parameters
+    ----------
+    s : str
+        The possible Roman numeral.
+
+    Returns
+    -------
+    result : bool
+        `True` if the `str` input is a valid Roman numeral, and `False`
+        if it is not.
+
+    Raises
+    ------
+    TypeError
+        If the argument is not a `str`.
+
+    See Also
+    --------
+    to_roman
+
+    from_roman
+
+    Examples
+    --------
+    >>> is_roman_numeral("CXVII")
+    True
+    >>> is_roman_numeral("42")
+    False
+
+    """
+    if not isinstance(s, str):
+        raise TypeError("Only strings may be tested ")
+    result = bool(_romanNumeralPattern.match(s))
+    return result
```

Comment:
Oh wow, you weren't joking when you said this was a small module. There's literally more *glorious numpydoc* in here than code, though if I remember correctly... well procrastinated, @namurphy ? :smile: